pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -158,6 +158,7 @@ func GetLogCollector(ctx context.Context, cli client.Client) (*operatorv1.LogCol
 		return nil, err
 	}
 
+	fillDefaults(instance)
```

Comment:
I don't think we can do the fillDefaults in this function because we need the "raw" instance in the Reconcile to save as preDefaultPatchFrom. Otherwise the reconcile will not see the that we're adding the CollectProcessPath and NOT patch the resource with the default.

Reply:
I also think in the Reconcile we need to move the patching of the LogCollector resource earlier, especially now since the Core controller will be reading the resource.
Please also as this move I think you will need to move https://github.com/tigera/operator/pull/1394/files#diff-2b8c59da3e46730154a614a6d0ea55cfc27b17106a1e4bcc3fc8c12f794e8571R351-R358 the defaulting of  instance.Spec.AdditionalStores.Syslog.LogTypes into the fillDefaults function.
Hopefully this all makes sense, let me know if it doesn't.